src/main/java/model/Money.java

```diff
@@ -0,0 +1,61 @@
+package model;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import util.NumberFormatStringParser;
+
+public class Money {
+    private static final int PRICE_AMOUNT_PER_LOTTO = 1000;
+    public static final Money ZERO = new Money(BigDecimal.ZERO);
+
+    private final BigDecimal amount;
+
+    public Money(int amount) {
+        this.amount = BigDecimal.valueOf(amount);
+    }
+
+    private Money(BigDecimal amount) {
+        this.amount = amount;
+    }
```

Comment:
하나를 주 생성자로 만들고 나머지는 부 생성자로 만드는 건 어떨까요?

```suggestion
    public Money(int amount) {
        this(BigDecimal.valueOf(amount));
    }

    private Money(BigDecimal amount) {
        this.amount = amount;
    }
```